tools/fork-network/src/cli.rs

```diff
@@ -416,15 +416,34 @@ impl ForkNetworkCommand {
         let storage = open_storage(&home_dir, near_config).unwrap();
         let store = storage.get_hot_store();
 
-        tracing::info!("Delete unneeded columns in the original DB");
-        let mut update = store.store_update();
-        for col in DBCol::iter() {
-            match col {
-                DBCol::DbVersion | DBCol::Misc | DBCol::State | DBCol::FlatState => {}
-                _ => update.delete_all(col),
+        let store_path = home_dir
+            .join(near_config.config.store.path.clone().unwrap_or_else(|| PathBuf::from("data")));
+        let temp_store_home = store_path.join("temp-trimmed-db");
+        if std::fs::metadata(&temp_store_home).is_ok() {
+            anyhow::bail!("Temp trimmed DB already exists; please delete temp-trimmed-db in the data directory first");
+        }
+        let temp_storage = open_storage(&temp_store_home, near_config).unwrap();
+        let temp_store = temp_storage.get_hot_store();
+        let temp_store_path = temp_store_home.join("data");
+
+        trim_database::trim_database(store, &near_config.genesis.config, temp_store)?;
```

Comment:
I'm getting this error:

```
thread 'main' panicked at tools/fork-network/src/cli.rs:148:88:
called `Result::unwrap()` on an `Err` value: MissingTrieValue(TrieStorage, 11111111111111111111111111111111)
```

to reproduce, apply the diff and run the two commands I posted in this comment: https://github.com/near/nearcore/pull/11310#discussion_r1609211659

I'm not sure exactly what is happening, but the error is in the call to `prepare_memtrie_state_trimming()` in `trim_database()`

Reply:
Oh. I think I know what happened, let me fix that.

Reply:
Hmm I have some trouble running the commands and I'm not sure if I did something wrong or something else, but I uploaded a fix that I think addresses the issue (handling empty tries); do you mind trying again?

Reply:
ah now i'm getting this:

```
thread 'main' panicked at tools/fork-network/src/cli.rs:148:88:
called `Result::unwrap()` on an `Err` value: MissingTrieValue(TrieStorage, 6Mswb4EnCzycL7n6bMEGaE87nnEboTj7K2BS3GyJjZbS
```

actually I dont know why I recommended doing the whole thing with the local mocknet stuff. You can reproduce this by just first running `python3 tests/sanity/transactions.py`

and then after that test is done, write something like this to `/tmp/validators.json`:

```
[{"account_id": "node0", "public_key": "ed25519:sqYjRhP3P1vzYY2nA7w2hCTZoA8hknuXpSeaReRxTKf", "amount": "1000000000000000000000000000000000"}, {"account_id": "node1", "public_key": "ed25519:4WFsnnwUj69smErNuidUEfeJs3fpHCQUJNsYzeghQVyT", "amount": "1000000000000000000000000000000000"}]
```

then just run the sequence of `fork-network` commands:

```
neard --home ~/.near/test4_finished fork-network init
neard --home ~/.near/test4_finished fork-network amend-access-keys
neard --home ~/.near/test4_finished fork-network set-validators --validators /tmp/validators.json
neard --home ~/.near/test4_finished fork-network finalize
```